shared/actions/push.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+import * as Constants from '../constants/push'
+
+// $FlowIssue tell flow about this module
+import * as PushNotifications from 'react-native-push-notification'
+
+import {apiserverPostRpcPromise} from '../constants/types/flow-types'
+
+import {call, put, select} from 'redux-saga/effects'
+import {takeEvery, takeLatest} from 'redux-saga'
+
+import type {SagaGenerator} from '../constants/types/saga'
+import type {TypedState} from '../constants/reducer'
+
+import type {PushNotification, PushNotificationAction, PushPermissionsPromptAction, PushPermissionsRequestAction, PushPermissionsRequestingAction, PushTokenAction, SavePushTokenAction, TokenType, UpdatePushTokenAction} from '../constants/push'
+
+export function permissionsRequest (): PushPermissionsRequestAction {
+  return {type: Constants.permissionsRequest, payload: undefined}
+}
+
+function * permissionsRequestSaga (): SagaGenerator<any, any> {
+  try {
+    yield put({type: Constants.permissionsRequesting, payload: true})
+
+    console.log('Requesting permissions')
+    const permissions = yield call(() => { return PushNotifications.requestPermissions() })
+    console.log('Permissions:', permissions)
+    // TODO(gabriel): Set permissions we have in store, might want it at some point?
+  } finally {
+    yield put({type: Constants.permissionsRequesting, payload: false})
+    yield put({type: Constants.permissionsPrompt, payload: false})
+  }
+}
+
+export function permissionsRequesting (enabled: boolean): PushPermissionsRequestingAction {
+  return {type: Constants.permissionsRequesting, payload: enabled}
+}
+
+export function permissionsPrompt (enabled: boolean): PushPermissionsPromptAction {
+  return {type: Constants.permissionsPrompt, payload: enabled}
+}
+
+export function pushNotification (notification: PushNotification): PushNotificationAction {
+  return {type: Constants.pushNotification, payload: notification}
+}
+
+function * pushNotificationSaga (notification: PushNotification): SagaGenerator<any, any> {
+  console.warn('Push notification:', notification)
+}
+
+export function pushToken (token: string, tokenType: TokenType): PushTokenAction {
```

Comment:
🐑 Order this file by putting the action creators like these at the top, and sagas at the bottom. together